package/src/lib/components/tour-overlay/shapes/RectShape.component.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { memo, useEffect, useMemo, useRef } from "react";
+import isEqual from "react-fast-compare";
+import { Animated } from "react-native";
+import { Rect } from "react-native-svg";
+
+import { ShapeProps } from "./shape.types";
+
+const AnimatedRect = Animated.createAnimatedComponent(Rect);
+
+export const RectShape = memo<ShapeProps>(props => {
+  const { motion, padding, setReference, spot, useNativeDriver } = props;
+
+  const width = useMemo((): number => {
+    return spot.width + padding;
+  }, [spot.width, padding]);
+
+  const height = useMemo((): number => {
+    return spot.height + padding;
+  }, [spot.height, padding]);
+
+  const x = useMemo((): number => {
+    return spot.x - ((width - spot.width) / 2);
+  }, [spot.x, spot.width, width]);
+
+  const y = useMemo((): number => {
+    return spot.y - ((height - spot.height) / 2);
+  }, [spot.y, spot.height, height]);
+
+  const size = useRef(new Animated.ValueXY({ x: width, y: height }, { useNativeDriver }));
+  const origin = useRef(new Animated.ValueXY({ x, y }, { useNativeDriver }));
+  const opacity = useRef(new Animated.Value(0, { useNativeDriver }));
+
+  useEffect(() => {
+    const transition = (): Animated.CompositeAnimation => {
```

Comment:
We are using this same transition method in both shapes, and i think we plan to make more shapes. Wouldn't be useful to brin this method into a custom hook or maybe a general method that can be reused in both components to avoid duplication. 😬

Reply:
Good catch! The animations are not 100% equal, but they are so similar that we can create a helper abstraction for them 🚀